src/meta/src/hummock/compaction/manual_compaction_picker.rs

```diff
@@ -87,32 +86,66 @@ impl CompactionPicker for ManualCompactionPicker {
             .collect();
 
         for table in &level_table_infos {
-            if level_handlers[level].is_pending_compact(&table.id) {
+            if level_handlers[select_level].is_pending_compact(&table.id) {
                 continue;
             }
 
-            let overlap_files = self
-                .overlap_strategy
-                .check_base_level_overlap(&[table.clone()], &levels[target_level].table_infos);
-
-            if overlap_files
-                .iter()
-                .any(|table| level_handlers[target_level].is_pending_compact(&table.id))
-            {
-                continue;
+            if select_level != target_level {
+                // to handle L0 sub-compaction
+                let overlap_files = self
+                    .overlap_strategy
+                    .check_base_level_overlap(&[table.clone()], target_tables);
+
+                if overlap_files
+                    .iter()
+                    .any(|table| level_handlers[target_level].is_pending_compact(&table.id))
+                {
+                    continue;
+                }
             }
 
             select_input_ssts.push(table.clone());
         }
 
+        let target_input_ssts = self
```

Comment:
For manual l0-sub-compaction, we can support the whole l0 compaction. for overlap_strategt now, we will select all target level (l0) file as target_input_ssts. 

shall we need to handle this case ?

Reply:
> > We shall also handle L0 case here, shan't we?
> 
> For manual l0-sub-compaction, we can support the whole l0 compaction. for overlap_strategt now, we will select all target level (l0) file as target_input_ssts.
> 
> shall we need to handle this case ?

even though we hold the selection logic here, we still need to merge 2 input_level(L0 and L0) into one, rather than split them into 2 'input level's as if they are in different level

Reply:
fix it cc @soundOfDestiny